### PR TITLE
PC-9660 Release channel in terraform refactor and improvements

### PR DESCRIPTION
**Pull request by dawidwisn**

### CODE REVIEW

File: nobl9/release_channel_schema.go [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+const releaseChannel = "release_channel"
+
+func schemaReleaseChannel() *schema.Schema {
+	return &schema.Schema{
+		Type:        schema.TypeString,
+		Optional:    true,
+		Description: "Release channel of the created datasource [stable/beta]",
```
Comment (nieomylnieja) [inline nobl9/release_channel_schema.go:14]: maybe add `Default` option, pointing at `stable`? It'd be aligned with what we do at backend --> setting stable channel when none was provided

File: nobl9/release_channel_schema.go [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+	}
+}
+
+func marshalReleaseChannel(d *schema.ResourceData) string {
+	return d.Get(releaseChannel).(string)
+}
```
Comment (nieomylnieja) [inline nobl9/release_channel_schema.go:20]: I'd parse the channel here and verify it's valid:

```suggestion
func marshalReleaseChannel(d *schema.ResourceData) (string, error) {
	rc, ok := d.Get(releaseChannel).(string)
	if !ok {
		return "", nil
	}
	if _, err := v1alpha.ParseReleaseChannel(rc); err != nil {
		return "", err
	}
	return rc, nil
}
```

File: nobl9/release_channel_schema.go (hunk 1 above)
Comment (dawidwisn) [reply nobl9/release_channel_schema.go:14]: On the backend, we can customize the default release channel for different organizations, so I would not put default values here.

Comment (nieomylnieja) [reply nobl9/release_channel_schema.go:17]: oh, didn't know that, makes sense :)

**Review by nieomylnieja — APPROVED** [review]

**Pull request merged**